sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -78,13 +87,44 @@ impl Default for InitialNodeConfig {
     }
 }
 
+fn check_sha256_sum(data: &Vec<u8>, sum: &str) -> bool {
```

Comment:
It feels this function does two things, would it be simpler to just return the sha256 and let the caller do the comparison? Also a panic on failure seems excessive!